lib/helpers/errorHandler.js

```diff
@@ -20,14 +20,20 @@ export default [
             if (command.name !== 'element' && command.name !== 'elements') {
                 continue
             }
-            if (command.name === 'element' && (!command.result[0].value || command.result[0].value.ELEMENT !== failingCommand.args[0])) {
+
+            if (command.name === 'element' && (!command.result || !command.result.value || command.result.value.ELEMENT !== failingCommand.args[0])) {
                 continue
             }
 
-            for (let result of command.result.value) {
-                if (result.ELEMENT === failingCommand.args[0]) {
-                    commandToRepeat = this.commandList[i - 1]
-                    break
+            if (command.result) {
+                // Ensure an array to simply the below logic
+                const results = Array.isArray(command.result.value) ? command.result.value : command.result.value !== null ? [command.result.value] : []
```

Comment:
@ebbe-brandstrup My solution here was a little different. Your solution is basically throwing out the possibility of retrying on any command that uses element rather than elements. My solution was not to ensure an array but simple check if we had an the result of an element command and if it was, check to see if that result was the elements we were looking for and set commandToRepeat there then break. You can see mine at https://github.com/CTaylor1118/webdriverio/blob/master/lib/helpers/errorHandler.js.